README.md

```diff
@@ -1,2 +1,150 @@
+conforms to the core [DID-CORE] specification. The method can be used independent of any central source of truth, and is intended for bootsrapping secure interaction between two parties that can span across arbitary number of devices. It is suitable for long sessions that need to operateing under network partitions.
+
+## Status of This Document
+
+This document is draft of a potential specification. It has no official standing of any kind and does not represent the support or consensus of any standards organization.
+
+## Intorduction
+
+### Overview
+
+> This section is non-normative.
+
+Most documentation about decentralized identifiers (DIDs) describes them as identifiers that are rooted in a public source of truth like a blockchain, a database, or similar. This publicness lets arbitrary parties resolve the DIDs to an endpoint and keys. It is an important feature for many use cases.
+
+However, the vast majority of interactions between people, organizations, and things have simpler requirements. When Alice(Corp|Device) and Bob want to interact, there are exactly and only 2 parties in the world who should care: Alice and Bob. Instead of arbitrary parties needing to resolve their DIDs, only Alice and Bob do.
+
+One to one interactions are excellent fit for [did:key] identifiers, however they suffer from key discovery problem and introduce additional challenges when interaction sessions span across more than two devices.
+
+Mailto DIDs are designed to be used in conjunction with [did:key] and facilitate bootstrapping sessions between two parties that span across multiple devices.
+
+Mailto DIDs are more accessible alternative to [did:web] and [did:dns] because a lot more people have an email address than there are people with [did:web] or [did:dns] identifier or skills to acquire one.
+
+Mailto [did document]s are also verifiable offline, when [domain key][] of the email address is known, implying significantly less network requests than with most other DIDs.
+
+## The `did:mailto` Format
+
+The format for the `did:key` method conforms to the [DID-CORE] specification and is encoding of the [email address]. It consists of the `did:mailto:` prefix, followed by the `domain` part of an email address, `:` character and percent encoded `local-part` of the email address.
+
+The ABNF definition can be found below. The formal rules describing valid `domain-name` syntax are described in [RFC1035], [RFC1123], [RFC2181]. The `domain-name` and `user-name` corresponds to `domain` and `local-part` respectively of the email address described in [RFC2822]. All "mailto" DIDs MUST conform to the DID Syntax ABNF Rules.
+
+```abnf
+did       = "did:mailto:" domain-name ":" user-name
+user-name = 1*idchar
+idchar    = ALPHA / DIGIT / "." / "-" / "_" / pct_enc
+pct_enc   = "%" HEXDIG HEXDIG
+```
+
+##### EXAMPLE 1. A jsmith@example.com
+
+```
+did:mailto:example.com:jsmith
+```
+
+##### EXAMPLE 2 tag+alice@web.mail
+
+```
+did:mailto:web.mail:tag%2Balice
```

Comment:
@blaine and I talked about the %-encoding a bit yesterday. I don't think the ugly `%`s are the end of the world, buthave you considered making this either a multibase or a hash?

A hash seems strange at first, but since you will have that info in the DKIM or similar later, you can validate it at that time. The downside is that it's harder for a human to track visually, but a fringe benefit is that you don't have to worry about encoding and they're constant size.

Reply:
> @blaine and I talked about the %-encoding a bit yesterday. I don't think the ugly `%`s are the end of the world, buthave you considered making this either a multibase or a hash?

I was mostly bothered about this in because it would turn `@` into `%40` which is why I end up moving username into the tail of the did. Otherwise I think most email addresses won't actually encounter this so probably not a big deal ?

I have thought about hashes, but mostly for the user privacy, but then realized it probably won't provide much. At the moment I think %-encoding a reasonable compromise and seems to be expected approach for DIDs, but I'm fine taking it other way.

> A hash seems strange at first, but since you will have that info in the DKIM or similar later, you can validate it at that time. The downside is that it's harder for a human to track visually, but a fringe benefit is that you don't have to worry about encoding and they're constant size.

I would prefer to not to with DKIM assumption, because we already have a [system in place](https://github.com/web3-storage/specs/blob/main/w3-account.md#email-validation) that adds did:mailto support without it, which I think will continue to be useful even when we implement DKIM support.